crates/matrix-sdk-store-encryption/src/lib.rs

```diff
@@ -603,89 +525,12 @@ impl StoreCipher {
     /// ```
     pub fn decrypt_value<T: DeserializeOwned>(&self, value: &[u8]) -> Result<T, Error> {
         let value: EncryptedValue = serde_json::from_slice(value)?;
-        self.decrypt_value_typed(value)
-    }
-
-    /// Decrypt a value after it was fetched from the key/value store.
-    ///
-    /// A value can be encrypted using the
-    /// [`StoreCipher::encrypt_value_typed()`] method. Lower level method to
-    /// [`StoreCipher::decrypt_value_typed()`]
-    ///
-    /// # Arguments
-    ///
-    /// * `value` - The EncryptedValue of a value that should be decrypted.
-    ///
-    /// The method will deserialize the decrypted value into the expected type.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// # let example = || {
-    /// use matrix_sdk_store_encryption::StoreCipher;
-    /// use serde_json::{json, value::Value};
-    ///
-    /// let store_cipher = StoreCipher::new()?;
-    ///
-    /// let value = json!({
-    ///     "some": "data",
-    /// });
-    ///
-    /// let encrypted = store_cipher.encrypt_value_typed(&value)?;
-    /// let decrypted: Value = store_cipher.decrypt_value_typed(encrypted)?;
-    ///
-    /// assert_eq!(value, decrypted);
-    /// # anyhow::Ok(()) };
-    /// ```
-    pub fn decrypt_value_typed<T: DeserializeOwned>(
```

Comment:
Removing this removes the abililty to use the `StoreCipher` with more efficient serialization formats, which at least one user used in the past: https://github.com/matrix-org/matrix-rust-sdk/pull/688.

Also, we wanted to remove the other, JSON methods: https://github.com/matrix-org/matrix-rust-sdk/issues/718.

While I understand the baroqueness argument and the poor naming of the methods I don't think I'd want to remove this functionality.

Reply:
> Removing this removes the ability to use the `StoreCipher` with more efficient serialization formats, which at least one user used in the past: #688.

Well, you can use `encrypt_value_data` instead?

> Also, we wanted to remove the other, the JSON based, methods: #718.
> 
> While I understand the baroqueness argument and the poor naming of the methods I don't think I'd want to remove this functionality.

Well, I'd quite like to get rid of `encrypt_value` and `decrypt_value` as well; I just didn't get that far.

Per my note on #718, I'm slightly confused about what it's asking for. In particular I note that https://github.com/matrix-org/matrix-rust-sdk/pull/3086 deprecated `decrypt_value_typed`. Has your thinking changed here?

Reply:
> Well, you can use encrypt_value_data instead?

Ah indeed, we still have that one. Then I don't have any objections.